TodoWithControllers/Program.cs

```diff
@@ -1,24 +1,26 @@
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.DependencyInjection;
+using System.Threading.Tasks;
 
-namespace Todos
+namespace TodoWithControllers
 {
-    class Program
+    public class Program
     {
-        static async Task Main(string[] args)
+        private static async Task Main(string[] args)
         {
             var builder = WebApplication.CreateBuilder(args);
 
             builder.Services.AddDbContext<TodoDbContext>(options => options.UseInMemoryDatabase("Todos"));
             builder.Services.AddControllers();
 
             var app = builder.Build();
+            if (args?.Length > 0)
```

Comment:
Ugh, no.

Reply:
Have no idea send request to what address when run `Program.cs` via reflection.
```csharp
var methodInfo = type.GetMethod("Main", BindingFlags.NonPublic | BindingFlags.Static);
const string port = "5001";
string[] args = { port };
methodInfo.Invoke(null, new[] { args });
```
or something like this:

```csharp
var directoryInfo = Directory.GetParent(Directory.GetCurrentDirectory());
var solutionDirectory = $"{directoryInfo.Parent.Parent.Parent.Parent.FullName}";
var projectName = type.FullName.Split('.')[0];
var launchSettingsPath = $"{solutionDirectory}\\{projectName}\\Properties\\launchSettings.json";

using var file = File.OpenText(launchSettingsPath);
using var reader = new JsonTextReader(file);
dynamic o2 = JToken.ReadFrom(reader);

var port = o2.iisSettings.iisExpress.sslPort;
Client = new HttpClient { BaseAddress = new Uri($"https://localhost:{port}") };
```

Reply:
Tip of the day:

- `ASPNETCORE_URLS`
- [Basic tests with the default WebApplicationFactory](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-3.1#basic-tests-with-the-default-webapplicationfactory)

Reply:
@Carl-Hugo thanks for tip of day. How about `--urls` command-line argument.
```csharp
string[] args = { "urls=https://localhost:5002" };
methodInfo.Invoke(null, new[] { args });
```

Reply:
> @Carl-Hugo thanks for the tip. How about `--urls` command-line argument.
> 
> ```cs
> string[] args = { "urls=https://localhost:5002" };
> methodInfo.Invoke(null, new[] { args });
> ```

As far as I know, they do the same thing; `args` being the last "settings override".